src/librustdoc/clean/mod.rs

```diff
@@ -1084,9 +1084,10 @@ impl GenericBound {
 
     fn get_trait_type(&self) -> Option<Type> {
```

Comment:
The existence of a trait path as a value of `Type` was a mistake, IMO.

If you move `DefId` into `Path`, you can replace
https://github.com/rust-lang/rust/blob/237bf3244fffef501cf37d4bda00e1fce3fcfb46/src/librustdoc/clean/mod.rs#L1196-L1201
With `PolyTrait { trait_: path, generic_params: late_bounds }`.

Reply:
Oh, you'd need this to take a `Def`, too (which can replace `trait_did`):
https://github.com/rust-lang/rust/blob/237bf3244fffef501cf37d4bda00e1fce3fcfb46/src/librustdoc/clean/mod.rs#L1153-L1154